modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeTopologyService.java

```diff
@@ -160,9 +184,25 @@ public ClusterNode getByConsistentId(String consistentId) {
      * @param member ScaleCube's cluster member.
      * @return Cluster node.
      */
-    private static ClusterNode fromMember(Member member) {
+    private static ClusterNode fromMember(Member member, @Nullable NodeMetadata nodeMetadata) {
         var addr = new NetworkAddress(member.address().host(), member.address().port());
 
-        return new ClusterNode(member.id(), member.alias(), addr);
+        return new ClusterNode(member.id(), member.alias(), addr, nodeMetadata);
+    }
+
+
+    /**
+     * Deserializes the given {@link ByteBuffer} to a {@link NodeMetadata}.
+     *
+     * @param buffer ByteBuffer to deserialize.
+     * @return NodeMetadata or null if something goes wrong.
+     */
+    private static NodeMetadata deserializeMetadata(ByteBuffer buffer) {
+        try {
+            return (NodeMetadata) METADATA_CODEC.deserialize(buffer);
```

Comment:
`buffer` can be null, this only works because you catch the NPE below. Let's add an explicit check